supermarket-intermediate/app.py

```diff
@@ -0,0 +1,319 @@
+    clear_screen()
+    for key in customers:
+        print(customers[key])
+        print('############################\n\n')
+    input('\n\nPress enter key to continue...')
+
+
+def list_registers_page(registers):
+    # Printing all existing registers to the user
+    clear_screen()
+    print(
+        """
+############################
+Available registers
+############################
+"""
+    )
+    for key in registers:
+        print(key)
+    input('\n\nPress enter key to continue...')
+
+
+def print_summary(registers):
+    # Printing summary of all registers to the user
+    clear_screen()
+    for key in registers:
+        print(registers[key])
+        print('############################')
+
+    input('\n\nPress enter key to continue...')
+
+
+def shopping_page(customers: dict, registers: dict):
+    # Printing 'shopping page', reading all the necessary input from user
+    # and takes care of all the user actions inside the shop page
+
+    clear_screen()
+
+    # Asking the user to enter a customer name
+    current_customer = input('Enter your customer name: ').lower()
+    if current_customer not in customers.keys():
+        print("Customer not exists, you can create new customer in the main menu.")
+        input('\n\nPress enter key to continue...')
+        return
+
+    # Asking the user to enter the name of register hey wants to buy from
+    current_register = input(
+        "Enter register name to checkout to (Leave empty to see available registers): ")
+    while current_register not in registers.keys():
+        clear_screen()
+        if current_register == '':
+            list_registers_page(registers)
+            current_register = input("Enter register name to checkout to: ")
+        else:
+            current_register = input(
+                "Register not found, enter register name to checkout to (Leave empty to see available registers): ")
+
+    # Entering the shop
+    while True:
+        print_shop(customers[current_customer])
+        user_action = input('Enter Your choice: ')
+        while not (user_action.isdigit() and 0 < int(user_action) < 6):
+            print_shop(customers[current_customer])
+            user_action = input('Invalid input! Enter Your choice again: ')
+
+        user_action = int(user_action)
+
+        # User adds a product to his cart
+        if user_action == 1:
+            clear_screen()
+            product_name = input("Enter product name: ").lower()
+
+            product_quantity = input("Enter product quantity: ")
+            while not product_quantity.isdigit():
+                product_quantity = input("Enter a valid product quantity: ")
+
+            if product_name not in customers[current_customer].shopping_list.keys():
+                product_price = input("Enter product price: ")
+                while not re.fullmatch("[0-9]+(\.[0-9]+)?", product_price):
+                    product_price = input("Enter a valid product price: ")
+            else:
```

Comment:
if you have already this "product_name" you  dont combine them into "product_quantity"